_posts/en/newsletters/2020-06-17-newsletter.md

```diff
@@ -0,0 +1,170 @@
+    protocol they call WabiSabi that conceptually extends their protocol
+    with a technique adapted from [confidential transactions][].  This allows a
+    client to create a commitment to arbitrary output amounts and---without
+    revealing the amounts---prove that each amount is individually
+    within a specified range (e.g. 0.0001 BTC to 21 million BTC) and
+    that they collectively sum to a specified value.  The coordinator
+    uses this specified value to verify that the sum of the outputs the
+    client wants to create is equal to the sum of the inputs provided by
+    the client (minus fees).  The coordinator can then provide an
+    anonymous credential for each output commitment that allows the
+    client to later anonymously submit the revealed output to the
+    coordinator for inclusion in the transaction template.
+
+    Software that implements the protocol will be able to create
+    coordinated coinjoins---with privacy from the coordinator---that
+    allow the clients to select their output amounts, facilitating
+    experiments with non-equal value coinjoins (see [Newsletter
+    #79][news79 unequal coinjoin]) and payments made within a coinjoin
+    (either to third-parties not participating in the coinjoin or to
+    parties within a coinjoin).
+
+    The proposed protocol contains a significant number of differences
+    from Wasabi's current protocol (such as replacing blind signatures
+    with keyed-verification anonymous credentials), so its authors are
+    seeking review, criticism, and suggestions about how the protocol
+    can be used most effectively.
+
+## Changes to services and client software
+
+*In this monthly feature, we highlight interesting updates to Bitcoin
+wallets and services.*
+
+FIXME:bitschmidty
+
+## Notable code and documentation changes
+
+*Notable changes this week in [Bitcoin Core][bitcoin core repo],
+[C-Lightning][c-lightning repo], [Eclair][eclair repo], [LND][lnd repo],
+[Rust-Lightning][rust-lightning repo], [libsecp256k1][libsecp256k1 repo],
+[Bitcoin Improvement Proposals (BIPs)][bips repo], and [Lightning
+BOLTs][bolts repo].*
+
+- [BIPs #910][] Assigns [BIP85][] to Ethan Kosakovsky's Deterministic Entropy From
+  BIP32 Keychains proposal. BIP85 describes a super-keychain whose child keys
+  can be used to create traditional HD keychain seeds. See [Newsletter #93][news93 bip32 keychains]
+  for more details on this proposal.
+
+{% include references.md %}
+{% include linkers/issues.md issues="910" %}
+[coinpool post]: https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2020-June/017964.html
+[joinpool]: https://gist.github.com/harding/a30864d0315a0cebd7de3732f5bd88f0
+[joinmarket]: https://github.com/JoinMarket-Org/joinmarket-clientserver
+[wasabi]: https://wasabiwallet.io/
+[wabisabi post]: https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2020-June/017969.html
+[wabisabi paper]: https://github.com/zkSNACKs/WabiSabi/releases/latest/download/WabiSabi.pdf
```

Comment:
I _really_ didn't like that this downloaded on click. Do you think that the link to the mailing list post is enough?

Reply:
Sure, the mailing list post is enough and I'm open to removing the link, but we've linked to PDFs many times before, so I curious to understand what's different about this one?

    $ git grep '\.pdf' -- _posts/en/newsletters/ | wc -l
    49

Reply:
The others look to be hosted online while this downloads to my machine without warning. I don't have a problem with the .pdf format per se.

Reply:
Ah.  Yeah, your browser probably has a built-in preview for PDF files that set their MIME type correctly, but this document has its MIME type set as binary (octet-stream):

![2020-06-15-14_35_25_429352728](https://user-images.githubusercontent.com/61096/84693609-c30ffb80-af15-11ea-8a35-5115657a66f9.png)

Amusingly, the document also is automatically downloaded if you're viewing it in a local-browser-based preview (and this has indeed been the vector for some previous browser security vulnerabilities, though perhaps not as many as non-browser-based PDF vulnerabilities).

In any case, GitHub net setting the correct MIME type is a good enough reason to remove the link.  I did try playing with the `<a>` tag's `type` attribute but, as the HTML5 standard specifies, I'm unable to use this to override the server's MIME type.

Reply:
When I click on it with a non-recent Firefox (68.9) on Debian/PureOS, I get the usual "What should Firefox do with this file?" dialog asking if I'd like to open the pdf or download it. In my case, I think it depends on the browser settings.

Reply:
I get the same (also on Firefox), but I tested Chromium and it automatically downloaded to the file system.  By comparison, if I open https://bitcoincore.org/bitcoin.pdf in Chromium, it opens in Chromium's version of pdf.js, which I think is what @adamjonas would prefer happen.